README.md

```diff
@@ -1,196 +1,289 @@
+
+## Installation
+#### Using Ubuntu (tested on 14.04/16.04)
+Create a working directory. Select the /opt folder as working directory (in our case we are using use ~/dx)
+```
+mkdir ~/dx
+cd ~/dx
+```
+Install prerequisites
+```
+sudo apt-get install git 
+sudo apt-get install wget
+```
+Enable universe repositories
+```
+sudo apt-get install software-properties-common
+sudo apt-add-repository universe
+sudo apt-get update
+```
+Install pip and some essential packages
+```
+curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py
+sudo apt install python2.7
+sudo cp /usr/bin/python2.7 /usr/bin/python
+sudo python get-pip.py 
+sudo pip install -U setuptools 
+sudo pip install packaging
+make
+sudo apt-get install build-essential -y
+Install ansible
+git clone https://github.com/ansible/ansible.git
+cd ansible/
+make
+sudo make install
+```
+Download or move some test sequencing data in `/opt/seq` folder  
+
+Clone streaming repo
+```
+cd ~/dx
+git clone https://github.com/dnanexus-rnd/dx-streaming-upload.git
+```
+Create dx-upload-play.yml file inside the dx-streaming-folder. 
+###dx-upload-play.yml
+`dx-upload-play.yml`
+```YAML
+---
+- hosts: localhost
+  vars:
+    monitored_users:
+      - username: root
+        local_tar_directory: /opt/tmp
+        local_log_directory: /opt/log
+        monitored_directories:
+          - /opt/seq
+        dx_user_token: <token>
+    mode: debug
+    upload_project: project-id
+  roles:
+    -  dx-streaming-upload
+```
+Here are the instructions for token generation.  
+Launch the ansible-playbook  
+```
+sudo ansible-playbook dx-streaming-upload/dx-upload-play.yml
+```
+Give the right permission to cron
+```
+sudo cron
+```
+#### Using RedHat (tested on 7.4 and 7.5)
+Create a working directory. Select the /opt folder as working directory (in our case we are using use ~/dx)
+```
+mkdir ~/dx 
+cd ~/dx
+```
+Install prerequisites
+```
+sudo yum install git -y
+sudo yum install wget -y
+```
+Enable EPEL Repository for RH 7.*
+```
+wget http://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm 
+sudo rpm -ivh epel-release-latest-7.noarch.rpm 
+Install pip and some essential packages
+sudo yum install python-pip -y 
+sudo cp /usr/bin/python2.7 /usr/bin/python
+sudo pip install -U setuptools 
+sudo pip install packaging
+make
+sudo yum install gcc gcc-c++ kernel-devel -y
+```
+Install ansible
+```
+git clone https://github.com/ansible/ansible.git
+cd ansible/
+make
+sudo make install
+```
+Download or move some test sequencing data in `/opt/seq` folder 
+
+Clone streaming repository
+```
+cd ~/dx
+git clone https://github.com/dnanexus-rnd/dx-streaming-upload.git
+```
+Create dx-upload-play.yml file inside the dx-streaming-folder. Example given [here](#dx-upload-play.yml)  

```

Comment:
--> " Create `dx-upload-play.yml` file inside the dx-streaming-folder. Example given [here](#dx-upload-play.yml)."